src/main/java/fr/cnes/sonar/plugins/icode/rules/ICodeRulesDefinition.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of sonar-icode-cnes-plugin.
+ *
+ * sonar-icode-cnes-plugin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sonar-icode-cnes-plugin is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with sonar-icode-cnes-plugin.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package fr.cnes.sonar.plugins.icode.rules;
+
+import fr.cnes.sonar.plugins.icode.languages.Fortran77Language;
+import fr.cnes.sonar.plugins.icode.languages.Fortran90Language;
+import fr.cnes.sonar.plugins.icode.languages.ShellLanguage;
+import fr.cnes.sonar.plugins.icode.settings.ICodePluginProperties;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.server.rule.RulesDefinitionXmlLoader;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Specific i-Code rules definition provided by resource file.
+ *
+ * @author lequal
+ */
+public class ICodeRulesDefinition implements RulesDefinition {
+
+	/** Partial key for repository. **/
+	private static final String REPO_KEY_SUFFIX = "-rules";
+
+	/** Path to xml file in resources tree (shell rules). **/
+	public static final String PATH_TO_SHELL_RULES_XML = "/rules/icode-shell-rules.xml";
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor your code to get this URI from a customizable parameter. ([squid:S1075](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS1075))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonar-icode-plugin&pullRequest=20&issues=AWSnS0Oosk71k7lw6nRL&open=AWSnS0Oosk71k7lw6nRL)

[ISSUE_KEY]: AWSnS0Oosk71k7lw6nRL